internal/services/recommendation_poller.go

```diff
@@ -72,6 +73,7 @@ func requestAndSaveRecommendation(kafkaMsg types.RecommendationKafkaMsg, recomme
 		log.Errorf("unable to list recommendation for: %v", err)
 		return poll_cycle_complete
 	}
+	recommendationRequest.Inc()
```

Comment:
I think this should be inside the function definition of `Update_recommendations` there can be many reason that ROSOCP send recommendationRequest to kruize but code never reaches here to increment the counter. One example is if ROSOCP is unable to unmarshal the API response.

Reply:
I think we should update this counter post usage of `Update_recommendations`, if in case there are errors encountered, we can explicitly handle the same in the poller service/where ever we used the function.

Reply:
But then we won't be getting the exact count i.e. total number of times we hit the update_recommendation API. If you see the error block above which says `recommendation does not exist for timestamp` I think in this case as well we should increment the prometheus counter.

Reply:
Change pushed

Reply:
Change pushed